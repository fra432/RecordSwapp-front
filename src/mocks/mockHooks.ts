export const mockNavigate = jest.fn();

export const mockAppDispatch = jest.fn();
