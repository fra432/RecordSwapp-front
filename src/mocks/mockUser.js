export const mockUserLogin = {
  username: "fra432",
  image: "image",
};
