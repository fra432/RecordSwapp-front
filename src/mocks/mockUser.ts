export const mockUserLogin = {
  username: "fra432",
  image: "image",
  imageBackup: "image",
};

export const mockUsers = [
  {
    id: "1",
    username: "fra432",
    image: "",
    imageBackup: "",
    location: "Barcelona",
    genre: "Electronic",
  },
  {
    id: "2",
    username: "nico",
    image: "",
    imageBackup: "",
    location: "Barcelona",
    genre: "Electronic",
  },
];

export const mockAppDispatch = jest.fn();
