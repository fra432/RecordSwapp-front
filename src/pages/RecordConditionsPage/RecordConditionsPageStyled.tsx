import styled from "styled-components";

const RecordConditionsPageStyled = styled.div`
  position: relative;
  background-color: #fafafa;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 80%;

  .title {
    padding-top: 1rem;
    font-weight: 400;
    word-spacing: 0.5px;
  }

  .conditions {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 1rem;
    gap: 1rem;

    .condition-grade {
      font-size: 1.2rem;
      font-weight: 600;
    }
  }

  .icon-back {
    cursor: pointer;
    margin-bottom: 1rem;
  }
`;

export default RecordConditionsPageStyled;
