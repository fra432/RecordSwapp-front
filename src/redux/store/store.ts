import { configureStore } from "@reduxjs/toolkit";
import recordSlice from "../features/recordSlice";
import recordsReducer from "../features/recordsSlice";
import userReducer from "../features/userSlice";
import usersReducer from "../features/usersSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    users: usersReducer,
    records: recordsReducer,
    record: recordSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
