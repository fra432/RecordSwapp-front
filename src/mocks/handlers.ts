import { rest } from "msw";
import { mockRecords } from "./mockRecords";
import { mockUserLogin, mockUsers } from "./mockUser";

const mockToken = "token";

export const handlers = [
  rest.post(`${process.env.REACT_APP_API_URL}user/login`, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ token: mockToken }));
  }),

  rest.post(
    `${process.env.REACT_APP_API_URL}user/register`,
    (req, res, ctx) => {
      return res(
        ctx.status(201),
        ctx.json({
          new_user: {
            username: mockUserLogin.username,
          },
        })
      );
    }
  ),

  rest.get(`${process.env.REACT_APP_API_URL}users`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        usersCollection: mockUsers,
      })
    );
  }),

  rest.get(`${process.env.REACT_APP_API_URL}myCollection`, (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        records: mockRecords,
      })
    );
  }),

  rest.post(`${process.env.REACT_APP_API_URL}myCollection`, (req, res, ctx) => {
    return res(
      ctx.status(201),
      ctx.json({
        new_record: mockRecords[0],
      })
    );
  }),

  rest.delete(
    `${process.env.REACT_APP_API_URL}myCollection/1`,
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          deleted_record: mockRecords[0],
        })
      );
    }
  ),

  rest.delete(
    `${process.env.REACT_APP_API_URL}myCollection/4`,
    (req, res, ctx) => {
      return res(ctx.status(404));
    }
  ),

  rest.put(
    `${process.env.REACT_APP_API_URL}myCollection/edit/1`,
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          updatedRecord: {
            title: "Neptune's Lairs",
            artist: "Drexciya",
            year: "2002",
            genre: "Electronic",
            price: "30",
            conditions: "VG",
            youtube_url: "https://www.youtube.com/watch?v=tF9rKnOqWfk",
            image:
              "https://i.discogs.com/MfE22D_C9EA8XEvN62IeSEjazP2mkpss7bVtzp614fg/rs:fit/g:sm/q:40/h:300/w:300/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9SLTEzODY2/LTE1NDE0MzA3ODct/NzM3MC5qcGVn.jpeg",
            owner: "6294b3038ee0cb91581a8ce6",
            id: "1",
          },
        })
      );
    }
  ),

  rest.put(
    `${process.env.REACT_APP_API_URL}myCollection/edit/4`,
    (req, res, ctx) => {
      return res(ctx.status(404));
    }
  ),
];
