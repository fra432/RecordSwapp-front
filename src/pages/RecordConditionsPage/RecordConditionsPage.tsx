import RecordConditionsPageStyled from "./RecordConditionsPageStyled";
import { ImCancelCircle } from "react-icons/im";
import { useNavigate } from "react-router-dom";

const RecordConditionsPage = () => {
  const navigate = useNavigate();

  const goBack = () => {
    navigate(-1);
  };

  return (
    <RecordConditionsPageStyled>
      <h1 className="title">Record conditions rate</h1>
      <div className="conditions">
        <div className="conditions-description">
          <h4 className="condition-grade">Mint(M)</h4>
          <p className="description">
            Absolutely perfect in every way. Certainly never been played,
            possibly even still sealed. Should be used sparingly as a grade, if
            at all.
          </p>
        </div>
        <div className="conditions-description">
          <h4 className="condition-grade">Near Mint(NM)</h4>
          <p className="description">
            A nearly perfect record. A NM or M- record has more than likely
            never been played, and the vinyl will play perfectly, with no
            imperfections during playback. Many dealers won't give a grade
            higher than this implying (perhaps correctly) that no record is ever
            truly perfect. The record should show no obvious signs of wear.
          </p>
        </div>
        <div className="conditions-description">
          <h4 className="condition-grade">Very Good Plus(VG+)</h4>
          <p className="description">
            Generally worth 50% of the Near Mint value. A Very Good Plus record
            will show some signs that it was played and otherwise handled by a
            previous owner who took good care of it. Defects should be more of a
            cosmetic nature, not affecting the actual playback as a whole.
          </p>
        </div>
        <div className="conditions-description">
          <h4 className="condition-grade">Very Good(VG)</h4>
          <p className="description">
            Generally worth 25% of Near Mint value. Many of the defects found in
            a VG+ record will be more pronounced in a VG disc. Surface noise
            will be evident upon playing, especially in soft passages and during
            a song's intro and fade,.
          </p>
        </div>
        <div className="conditions-description">
          <h4 className="condition-grade">Good(G)</h4>
          <p className="description">
            Generally worth 10-15% of the Near Mint value. A record in Good or
            Good Plus condition can be played through without skipping. But it
            will have significant surface noise, scratches, and visible groove
            wear.
          </p>
        </div>
        <div className="conditions-description">
          <h4 className="condition-grade">Poor(P)</h4>
          <p className="description">
            Generally worth 0-5% of the Near Mint price. The record is cracked,
            badly warped, and won't play through without skipping or repeating.
          </p>
        </div>
      </div>
      <ImCancelCircle
        size={30}
        data-testid="icon-back"
        className="icon-back"
        onClick={goBack}
      />
    </RecordConditionsPageStyled>
  );
};

export default RecordConditionsPage;
