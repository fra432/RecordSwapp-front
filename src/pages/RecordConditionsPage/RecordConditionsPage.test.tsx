import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import reactTestRenderer from "react-test-renderer";
import { mockNavigate } from "../../mocks/mockHooks";
import store from "../../redux/store/store";
import RecordConditionsPage from "./RecordConditionsPage";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Given a RecordConditionsPage component function", () => {
  describe("When invoked and the users click on the close icon", () => {
    test("Then the useNavigate hooks should be called", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <RecordConditionsPage />
          </Provider>
        </BrowserRouter>
      );

      const goBackIcon = screen.getByTestId("icon-back");

      userEvent.click(goBackIcon);

      expect(mockNavigate).toHaveBeenCalled();
    });
  });

  describe("When invoked", () => {
    test("Then it should render correctly the RecordConditionsPage", () => {
      const recordConditionsPage = reactTestRenderer
        .create(
          <BrowserRouter>
            <Provider store={store}>
              <RecordConditionsPage />
            </Provider>
          </BrowserRouter>
        )
        .toJSON();

      expect(recordConditionsPage).toMatchSnapshot();
    });
  });
});
