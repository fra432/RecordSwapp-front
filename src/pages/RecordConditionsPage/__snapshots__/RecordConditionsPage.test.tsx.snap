// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RecordConditionsPage component function When invoked Then it should render correctly the RecordConditionsPage 1`] = `
<div
  className="sc-bczRLJ gmAJMU"
>
  <h1
    className="title"
  >
    Record conditions rate
  </h1>
  <div
    className="conditions"
  >
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Mint(M)
      </h4>
      <p
        className="description"
      >
        Absolutely perfect in every way. Certainly never been played, possibly even still sealed. Should be used sparingly as a grade, if at all.
      </p>
    </div>
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Near Mint(NM)
      </h4>
      <p
        className="description"
      >
        A nearly perfect record. A NM or M- record has more than likely never been played, and the vinyl will play perfectly, with no imperfections during playback. Many dealers won't give a grade higher than this implying (perhaps correctly) that no record is ever truly perfect. The record should show no obvious signs of wear.
      </p>
    </div>
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Very Good Plus(VG+)
      </h4>
      <p
        className="description"
      >
        Generally worth 50% of the Near Mint value. A Very Good Plus record will show some signs that it was played and otherwise handled by a previous owner who took good care of it. Defects should be more of a cosmetic nature, not affecting the actual playback as a whole.
      </p>
    </div>
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Very Good(VG)
      </h4>
      <p
        className="description"
      >
        Generally worth 25% of Near Mint value. Many of the defects found in a VG+ record will be more pronounced in a VG disc. Surface noise will be evident upon playing, especially in soft passages and during a song's intro and fade,.
      </p>
    </div>
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Good(G)
      </h4>
      <p
        className="description"
      >
        Generally worth 10-15% of the Near Mint value. A record in Good or Good Plus condition can be played through without skipping. But it will have significant surface noise, scratches, and visible groove wear.
      </p>
    </div>
    <div
      className="conditions-description"
    >
      <h4
        className="condition-grade"
      >
        Poor(P)
      </h4>
      <p
        className="description"
      >
        Generally worth 0-5% of the Near Mint price. The record is cracked, badly warped, and won't play through without skipping or repeating.
      </p>
    </div>
  </div>
  <svg
    className="icon-back"
    data-testid="icon-back"
    fill="currentColor"
    height={30}
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    version="1.1"
    viewBox="0 0 16 16"
    width={30}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
    />
    <path
      d="M10.5 4l-2.5 2.5-2.5-2.5-1.5 1.5 2.5 2.5-2.5 2.5 1.5 1.5 2.5-2.5 2.5 2.5 1.5-1.5-2.5-2.5 2.5-2.5z"
    />
  </svg>
</div>
`;
